src/Parameter.cpp

```diff
@@ -19,15 +19,15 @@ struct ParameterContents {
     const int dimensions;
     const std::string name;
     Buffer<> buffer;
-    uint64_t data;
+    uint64_t data = 0;
     int host_alignment;
     std::vector<BufferConstraint> buffer_constraints;
     Expr scalar_default, scalar_min, scalar_max, scalar_estimate;
     const bool is_buffer;
     MemoryType memory_type = MemoryType::Auto;
 
     ParameterContents(Type t, bool b, int d, const std::string &n)
```

Comment:
Reading through this PR, I'm not so sure this new check is a good thing. It splits up initialization of fields by a not-very-useful criterion so that you have to look in two places to discover what the initial values are for the members.